src/bpftrace.cpp

```diff
@@ -1754,8 +1755,9 @@ std::string BPFtrace::resolve_timestamp(uint32_t mode,
   snprintf(usecs_buf, sizeof(usecs_buf), "%06" PRIu64, us);
   auto fmt = std::regex_replace(raw_fmt, usec_regex, usecs_buf);
 
-  char timestr[config_.get(ConfigKeyInt::max_strlen)];
-  if (strftime(timestr, sizeof(timestr), fmt.c_str(), &tmp) == 0) {
+  uint64_t timestr_size = config_.get(ConfigKeyInt::max_strlen);
+  char *timestr = (char *)alloca(timestr_size);
```

Comment:
This one I would definitely put on the heap. `max_strlen` is a user-supplied number so could potentially be pretty big. We don't have any limits on what it can be set to either... I didn't know it was being used like this actually.

`std::vector` will at least catch the out-of-memory condition and throw an error.

Reply:
Already on it. :)

Reply:
Also done in the latest commit.